skimage/color/colorconv.py

```diff
@@ -804,6 +804,60 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create a RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        Alpha channel of the output image. It may be a scalar or an
+        array that can be broadcast to ``image``. If not specified it is
+        set to the maximum limit correspondin to ``image`` data type.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+        If ``alpha`` is a scalar and is not in ``image`` data type
+        limits.
+    ValueError
+        If ``alpha`` is an array and have a different shape then
+        ``image``.
+    ValueError
+        If ``alpha`` is an array and have a different data type then
+        ``image``.
+
+    """
+
+    arr = np.asarray(image)
```

Comment:
I think this is the job of a static type checker, or a linter. No need to always cast this to an array here.

Reply:
All right, but `image` has to be documented as `array`, or `ndarray` then... This also prevent for example from passing only one value or a small list of values...
The only thing preventing from using `asarray` is if we want to use `dask`, but does this function really need to be accelerated?
I think that the small overhead of `asarray` worth it...

Reply:
I agree that we don't really want to do this unless necessary.  That said, this problem is so pervasive across all of scikit-image that, for now, it's probably safer to make the asarray call (we don't generally support dask arrays, e.g.).